message_verifier.go

```diff
@@ -0,0 +1,272 @@
+package httpsignatures
+
+import (
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const (
+	digestSignatureStringHeader = "digest"
+)
+
+var (
+	keyIDRegex     = regexp.MustCompile(`keyId="([^"]+)"`)
```

Comment:
I think there are cases, where such regex approach would fail. For example `signature="asdfkeyId=",digest="zxcv"`

Reply:
how about adding `[ (",)]` before `keyId`? that will match either
`Signature keyId="abc"` or
`Signature algorithm="SHA-256",keyId="abc"`

Reply:
👍 I think it should work, would be nice to have tests for that. It would also apply to other regular expressions as well.

Reply:
done and added tests. Was a slight change to include start of string as an option - `(?:^| |(?:",))keyId="([^"]+)"`